copilot-chat-command.el

```diff
@@ -0,0 +1,938 @@
+
+;; customs
+(defcustom copilot-chat-list-added-buffers-only nil
+  "If non-nil, only show buffers that have been added to the Copilot chat list."
+  :type 'boolean
+  :group 'copilot-chat)
+
+(defcustom copilot-chat-ignored-commit-files
+  '("pnpm-lock.yaml" "package-lock.json" "yarn.lock" "poetry.lock"
+     "Cargo.lock" "go.sum" "composer.lock" "Gemfile.lock"
+     "requirements.txt" "*.pyc" "*.pyo" "*.pyd"
+     "*.so" "*.dylib" "*.dll" "*.exe"
+     "*.jar" "*.war" "*.ear"
+     "node_modules/" "vendor/" "dist/" "build/")
+  "List of file patterns to ignore when generating commit messages.
+These are typically large generated files like lock files or build artifacts
+that don't need to be included in commit message generation.
+Supports glob patterns like `*.lock' or `node_modules/'."
+  :type '(repeat string)
+  :group 'copilot-chat)
+
+;; Faces
+(defface copilot-chat-list-selected-buffer-face
+  '((t :inherit font-lock-keyword-face))
+  "Face used for selected buffers in copilot-chat buffer list."
+  :group 'copilot-chat)
+(defface copilot-chat-list-default-face
+  '((t :inherit default))
+  "Face used for unselected buffers in copilot-chat buffer list."
+  :group 'copilot-chat)
+
+;; Variables
+(defvar copilot-chat-list-buffer "*Copilot-chat-list*")
+(defvar copilot-chat-prompt-mode-map
+  (let ((map (make-keymap)))
+    (define-key map (kbd "C-c RET") 'copilot-chat-prompt-send)
+    (define-key map (kbd "C-c C-c") 'copilot-chat-prompt-send)
+    (define-key map (kbd "C-c C-q") (lambda()
+                                      (interactive)
+                                      (bury-buffer)
+                                      (delete-window)))
+    (define-key map (kbd "C-c C-l") 'copilot-chat-prompt-split-and-list)
+    (define-key map (kbd "C-c C-t") 'copilot-chat-transient)
+    (define-key map (kbd "M-p") 'copilot-chat-prompt-history-previous)
+    (define-key map (kbd "M-n") 'copilot-chat-prompt-history-next)
+    map)
+  "Keymap for Copilot Chat Prompt mode.")
+(defvar copilot-chat-list-mode-map
+  (let ((map (make-sparse-keymap)))
+    (define-key map (kbd "RET") 'copilot-chat-list-add-or-remove-buffer)
+    (define-key map (kbd "SPC") 'copilot-chat-list-add-or-remove-buffer)
+    (define-key map (kbd "C-c C-c") 'copilot-chat-list-clear-buffers)
+    (define-key map (kbd "g") 'copilot-chat-list-refresh)
+    (define-key map (kbd "q") (lambda()
+                                (interactive)
+                                (bury-buffer)
+                                (delete-window)))
+    map)
+  "Keymap for `copilot-chat-list-mode'.")
+(defvar copilot-chat--prompt-history nil
+  "Copilot-chat prompt history.")
+(defvar copilot-chat--prompt-history-position nil
+  "Current position in copilot-chat prompt history.")
+
+;; Functions
+(define-minor-mode copilot-chat-prompt-mode
+  "Minor mode for the Copilot Chat Prompt region."
+  :init-value nil
+  :lighter " Copilot Chat Prompt"
+  :keymap copilot-chat-prompt-mode-map)
```

Comment:
Why did you remove the hook ?

Reply:
`define-minor-mode` create hook by default.
The hook of original code is plain.
default is sufficient.